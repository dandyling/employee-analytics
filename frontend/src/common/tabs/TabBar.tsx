import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export interface TabHeader {
  tabName: string;
  tabIcon?: any;
}

interface Props {
  tabs: TabHeader[];
  onChange(index: number): void;
  selected: number;
}

export const TabBar = (props: Props) => {
  const { tabs, onChange, selected } = props;

  return (
    <div className="flex">
      {tabs.map((t, i) => {
        const isSelected = selected === i;
        const selectedStyle = isSelected ? "bg-opacity-20 bg-white" : "";

        const handleChange = () => {
          onChange(i);
        };

        return (
          <div key={`tab-${t.tabName}`}>
            <button
              onClick={handleChange}
              className={`flex items-center duration-500 px-4 py-2 transition-colors focus:outline-none rounded ${selectedStyle}`}
            >
              {t.tabIcon && (
                <div className="pr-2 text-lg">
                  <FontAwesomeIcon icon={t.tabIcon} onClick={console.log} />
                </div>
              )}
              <div className="text-sm uppercase">{t.tabName}</div>
            </button>
          </div>
        );
      })}
    </div>
  );
};
