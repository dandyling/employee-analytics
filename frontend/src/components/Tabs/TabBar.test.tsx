import { render, screen } from "@testing-library/react";
import React from "react";
import { TabBar, TabHeader } from "./TabBar";

const testTabs: TabHeader[] = [{ tabName: "Tab 1" }, { tabName: "Tab 2" }];

test("renders label on tabs", () => {
  render(<TabBar tabs={testTabs} onChange={() => {}} selected={0} />);
  const tabElement1 = screen.getByText(/Tab 1/i);
  expect(tabElement1).toBeInTheDocument();
  const tabElement2 = screen.getByText(/Tab 2/i);
  expect(tabElement2).toBeInTheDocument();
});

test("does not render when tabs is an empty array", () => {
  const { container } = render(
    <TabBar tabs={[]} onChange={() => {}} selected={0} />
  );
  expect(container.firstChild).toBeEmptyDOMElement();
});

test("renders correctly", () => {
  const wrapper = <TabBar tabs={testTabs} onChange={() => {}} selected={0} />;
  expect(wrapper).toMatchSnapshot();
});
