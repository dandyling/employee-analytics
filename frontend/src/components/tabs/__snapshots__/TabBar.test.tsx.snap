// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TabBar
  onChange={[Function]}
  selected={0}
  tabs={
    Array [
      Object {
        "tabName": "Tab 1",
      },
      Object {
        "tabName": "Tab 2",
      },
    ]
  }
/>
`;
